plugins/org.python.pydev/tests/org/python/pydev/editor/PyAutoIndentStrategyTest.java

```diff
@@ -1297,6 +1297,17 @@ public void testAutoSelf2() {
         assertEquals(expected, docCmd.text);
     }
 
+    public void testExitAutoSelf() {
+        strategy.setIndentPrefs(new TestIndentPrefs(false, 4, true));
+        String doc = ""
+                + "class A:\r\n" +
```

Comment:
Can you do a new test where the user does:

```
class A:
    def my__exit__
```

And see that it won't add it?.